sdk/core/core-amqp/src/requestResponseLink.ts

```diff
@@ -262,41 +261,50 @@ const getCodeDescriptionAndError = (props: any): NormalizedInfo => {
  */
 export function onMessageReceived(
   context: EventContext,
-  connection: Connection,
+  connectionId: string,
   responsesMap: Map<string, DeferredPromiseWithCallback>
 ): void {
   const message = context.message;
-  if (message) {
-    const responseCorrelationId = message.correlation_id;
-    if (responsesMap.has(responseCorrelationId as string)) {
-      const promise = responsesMap.get(
-        responseCorrelationId as string
-      ) as DeferredPromiseWithCallback;
-      promise.cleanupBeforeResolveOrReject();
+  if (!message) {
+    return;
```

Comment:
This might be worth a log message (I know you just shifted the code around).

If this happens then we're in some weird other-worldly circumstance, I imagine.

Reply:
Added `log` here and wherever I felt needed